Code/Framework/AzCore/AzCore/Metrics/IEventLogger.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <type_traits>
+#include <AzCore/base.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+#include <AzCore/std/limits.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/Casting/numeric_cast.h>
+
+namespace AZ::Metrics
+{
+    //! Simple hash structure based on DJB2a to generate event IDs at compile time
+    class EventNameHash
+    {
+    public:
+        constexpr explicit EventNameHash(AZStd::string_view name)
+        {
+            for (auto c : name)
+            {
+                m_hash = static_cast<uint32_t>(((static_cast<uint64_t>(m_hash) << 5) + m_hash) ^ c);
+            }
+        }
+
+        constexpr bool operator==(const EventNameHash& rhs) const
+        {
+            return m_hash == rhs.m_hash;
+        }
+
+        constexpr bool operator!=(const EventNameHash& rhs) const
+        {
+            return m_hash != rhs.m_hash;
+        }
+
+    private:
+        uint32_t m_hash{ 5381 }; // standard starting value for DJB2a hash
+    };
+
+
+    constexpr EventNameHash PrologEventHash("Prolog");
+    constexpr uint16_t EventBoundary = 8;
+
+
+    class IEventLogger
+    {
+    public:
+        struct LogHeader
+        {
+            int8_t m_4cc[4]{ 'A', 'Z', 'E', 'L' };  //!< 4CC to uniquely identify the data type. Defaults to 'AZEL'
```

Comment:
Special identifying bytes for a blob are usually called magic numbers. Is are these bytes fixed?

Reply:
Actually this is a file move. Not a modification.
It looks like I am missing the removal of the AzCore/Debug/IEventLogger.h
I'll fix that